### ISSUE TITLE
Type 'SwinjectStoryboardOption' does not conform to protocol 'ServiceKeyOption' 

### ISSUE THREAD

**Issue by Sega-Zero** [Swinject/SwinjectStoryboard#127]
Upgraded Swinject to 2.7.0 and `SwinjectStoryboard` now doesn't compile, with the error above.
I believe, there must be something like this in code:
```swift
internal struct SwinjectStoryboardOption: ServiceKeyOption {
    func hash(into: inout Hasher) {
        into.combine(self.controllerType)
    }
```

should I open a PR with a fix?

**Comment by AlexanderEBaranov** [comment]
when this problem will fixed ?

**Comment by dvshelley** [comment]
Here is the actual error.

error: type 'SwinjectStoryboardOption' does not conform to protocol 'ServiceKeyOption'
internal struct SwinjectStoryboardOption: ServiceKeyOption {
                ^
..... /Carthage/Checkouts/SwinjectStoryboard/Sources/SwinjectStoryboardOption.swift:12:17: note: do you want to add protocol stubs?
internal struct SwinjectStoryboardOption: ServiceKeyOption {
                ^

The cause of the error is the Swinject project moving to Swift 5 as part of their 2.7.0 release. Rolling back to Swinject 2.6.2 worked around the problem for me.

**Comment by adamayoung** [comment]
Is there an ETA on a fix?

**Comment by lygialiem** [comment]
I faced this issue. Just conform it and do not mofidy anything else in the Func which has just been added.

**Comment by Mustafa-Ezzat** [comment]
Please add the following method:
    func hash(into: inout Hasher) {
        into.combine(self.controllerType)
    }
in
SwinjectStoryboardOption.swift
then 
release

**Comment by Panajev** [comment]
Hello @jakubvano it seems like this fixes all the issues we had? Do you think it will go live/released soon?

**Comment by michzio** [comment]
Yeas i have experienced this in Xcode 11 in latest SwinjectStoryboard library

**Comment by lauramejiaarroyave** [comment]
T_T For those of us who use CI and we have the pods in the gitignore, what a problem, someone has news?

**Comment by Sega-Zero** [comment]
It appears that master branch head contains this code already, there's just no update on cocoapods. so, those who use a CI for now may specify branch in Podfile.
Hope @yoichitgy or @jakubvano will release it soon =)

**Comment by pallzoltan** [comment]
Clean checkout & build with carthage fails becuase it checks out commit [e687b47] tagged with `2.2.0`. @jakubvano, would you tag commit [0ca45c8] with `2.2.1`?

**Comment by lauramejiaarroyave** [comment]
`Closed`  Pero no hacen merge del PR. :(

**Comment by ChaosCoder** [comment]
Please release a new version (e.g. `2.2.1`) from `master`.

**Comment by sam-moshenko** [comment]
I think a cleaner fix is to go to `Podfile.lock` and look for `Swinject`, set it's version to 2.6.0 instead of 2.7.x
As soon as `SwinjectStoryboard` releases a fix, you can just update the `SwinjectStoryboard` and I believe it will automatically update `Swinject`

**Comment by gilroykilroy** [comment]
I respectfully ask is anyone paying attention anymore? This issue has existed for months already.

**Comment by dvshelley** [comment]
@sam-moshenko You are right. That would work around the problem until a new release for SwinjectStoryboard is created.

But I think the bigger question is from @gilroykilroy about whether or not there is going to be another SwinjectStoryboard release. The code to fix the problem has been checked into master for quite a while. I'm really unclear when a release was not created when the code was merged?

**Comment by matthiasnagel** [comment]
@jakubvano I just want to bring up this topic again. How can I help you releasing a new version with the fix?

**Comment by jcfrank77** [comment]
Is there any expected ETA for this fix?  Seems like a pretty simple change that would fix 2.7.0?

**Comment by teejayhh** [comment]
this is I still have this issue in 2021 the problem very much alive.

**Comment by sskjames** [comment]
Still facing this issue. Using this as a temporary workaround:

    pod 'Swinject', '~> 2.7.1'
    pod 'SwinjectStoryboard', :git => 'https://github.com/Swinject/SwinjectStoryboard.git', :commit => '0ca45c83a8aa398c153d8a036c95abb4343cfa0c'

**Comment by mapo80** [comment]
Can you fix this?

**Comment by mpdifran** [comment]
Sorry for the late reply here! I've just released version 2.2.1 which should have a fix for this issue included. 

I'm going to close this issue for now, please reopen if the issue isn't actually fixed!

**Comment by Bartson** [comment]
Hi, the problem still exists on v2.2.1 :/
